src/main/java/com/snowflake/kafka/connector/internal/JdbcProperties.java

```diff
@@ -0,0 +1,89 @@
+package com.snowflake.kafka.connector.internal;
+
+import java.util.Properties;
+
+/** Wrapper class for all snowflake jdbc properties */
+public class JdbcProperties {
+
+  /** All jdbc properties including proxyProperties */
+  private final Properties properties;
+  /** Proxy related properties */
+  private final Properties proxyProperties;
+
+  private JdbcProperties(Properties combinedProperties, Properties proxyProperties) {
+    this.properties = combinedProperties;
+    this.proxyProperties = proxyProperties;
+  }
+
+  public Properties getProperties() {
+    Properties copy = new Properties();
```

Comment:
Is there a reason for making a copy here?

Reply:
Super anti bug safety? I can remove it. I would not insist on keeping that.

Reply:
I reconsidered and removed the copy.